.github/workflows/publish.yml

```diff
@@ -26,12 +26,12 @@ jobs:
   publish:
     runs-on: ubuntu-24.04
     needs: [linter, tests]
+    if: startsWith(github.ref, 'refs/tags/')
```

Comment:
There is no need to execute some steps of the `publish` job. We can skip entirely the job if it's not a Git tag.